src/components/plot/plot-control-grid.tsx

```diff
@@ -0,0 +1,169 @@
+import '@isrd-isi-edu/deriva-webapps/src/assets/scss/_heatmap.scss';
+
+// hooks
+import { useEffect, useState } from 'react';
+import '/node_modules/react-resizable/css/styles.css';
+import '/node_modules/react-grid-layout/css/styles.css';
+// services
+import ChaiseSpinner from '@isrd-isi-edu/chaise/src/components/spinner';
+import { DataConfig, defaultGridProps } from '@isrd-isi-edu/deriva-webapps/src/models/plot';
+import { convertKeysSnakeToCamel, validateGridProps } from '@isrd-isi-edu/deriva-webapps/src/utils/string';
+import { LayoutConfig } from '@isrd-isi-edu/deriva-webapps/src/models/plot';
+import { setControlData, useControl } from '@isrd-isi-edu/deriva-webapps/src/hooks/control';
+import { Layouts, Responsive, WidthProvider } from 'react-grid-layout';
+import ChartWithEffect from '@isrd-isi-edu/deriva-webapps/src/components/plot/chart-with-effect';
+import UserControl from '@isrd-isi-edu/deriva-webapps/src/components/plot/user-control';
+
+
+
+export type PlotControlGridProps = {
+  config: DataConfig,
+};
+const ResponsiveGridLayout = WidthProvider(Responsive);
+/**
+ * 
+ * @param config config object for the given plot
+ * @returns 
+ */
+const PlotControlGrid = ({
+  config,
+}: PlotControlGridProps): JSX.Element => {
+
+  const [layout, setLayout] = useState<Layouts>({});
+  const [gridProps, setGridProps] = useState({});
+  const [dataOptions, setDataOptions] = useState<any>(null);
+  const [userControlsExists, setUserControlExists] = useState<boolean>(false);
+  const [userControlsReady, setUserControlReady] = useState<boolean>(false);
+  const [initialParams, setInitialParams] = useState<any>({
```

Comment:
Shouldn't this be using `templateParams` from the provider?

I think each place in this file where `setInitialParams` is called should be using `setTemplateParams` too. This component shouldn't be keeping track of this state when we are trying to have the state tracked in the provider.